frontend/src/components/HpView.jsx

```diff
@@ -6,39 +6,39 @@ const COLOR_YELLOW = new THREE.Color("yellow")
 const COLOR_ORANGE = new THREE.Color("orange")
 const COLOR_RED = new THREE.Color("red")
 
-export default function HpView({hp, yposition, width = 0.5, mode = "slider"})
+export default function HpView({hpFraction, position, width = 0.5, mode = "slider", modelHeight = 0.5})
```

Comment:
I don't think it makes sense to expose this if it's not used and the position is provided anyway.
```suggestion
export default function HpView({hpFraction, position, width = 0.5, mode = "slider"})
```

Reply:
```js
const position = [coordinates.x, coordinates.y + 0.5, coordinates.z];
...
        <HpView hpFraction={hp/maxHp} position={position} />
...
```

Reply:
The + 0.5 puts the hp bar above the enemy, rather than at their feet, but I was thinking that different units might want the bar at different heights.

Reply:
Agreed, that's why I had initially exposed `yposition` as a prop. I was hoping there would be a way for the `HpView` to inherit the remaining position details from the parent component but I hadn't experimented with that yet. It's equivalent to thinking of the hp bar as being in a model-relative coordinate space.

I think for that idea to work, we would have to wrap the `HpView` and `primitive` components in a `group` and apply the position to that instead.

Putting that idea aside for a moment, if we do end up passing position directly to `HpView` I feel the extra "modelHeight" prop is kind of redundant. That offset is not really relevant to the component itself and might as well be passed as part of the position instead.